Final Code

Nedkins opened the pull request:
Ready for review

Nedkins committed:
Final Code

Ready for review

Fnly23 commented on CMP1903_A1_2324/Die.cs:
```diff
@@ -6,18 +6,29 @@
 
```
potentially unecessary imports

KFANullSense replied:
(these are in the base provided code so I would not call them unnecessary)

Fnly23 commented on CMP1903_A1_2324/Die.cs:
The variables need to be written in _camelCase and camelCase for private and public

Fnly23 commented on CMP1903_A1_2324/Game.cs:
```diff
@@ -1,21 +1,51 @@
+            Console.WriteLine("Die 1: " + die1.CurrentValue);
+            Console.WriteLine("Die 2: " + die2.CurrentValue);
+            Console.WriteLine("Die 3: " + die3.CurrentValue);
+            Console.WriteLine("Total of the three dice rolls: " + total);
```
Could break this up to make it easier to read for people.

Fnly23 commented on CMP1903_A1_2324/Testing.cs:
More _camelCase and camelCase please.

Fnly23 reviewed (commented):
Overall looks very good and clean.

KFANullSense commented on CMP1903_A1_2324/Die.cs:
```diff
@@ -6,18 +6,29 @@
-         * and one method that rolls the die, returns and integer and takes no parameters.
-         */
+        // Stores the random object for number generation
+        private Random rng;
```
This should be _rng, as it is a private variable.

KFANullSense commented on CMP1903_A1_2324/Game.cs:
```diff
@@ -1,21 +1,51 @@
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+﻿using DieGame_Assessment_1;
```
I don't think this is needed? You've changed the namespace unnecessarily

KFANullSense commented on CMP1903_A1_2324/Game.cs:
```diff
@@ -1,21 +1,51 @@
+using System.Diagnostics;
 
-namespace CMP1903_A1_2324
+namespace DieGame_Assessment_1
```
This namespace change was not consistent throughout the project.

KFANullSense commented on CMP1903_A1_2324/Game.cs:
```diff
@@ -1,21 +1,51 @@
-         * rolls could be continous, and the totals and other statistics could be summarised for example.
-         */
+        // Stores the random object for number generation
+        private Random rng;
```
Again, should be _rng.

KFANullSense commented on CMP1903_A1_2324/Game.cs:
```diff
@@ -1,21 +1,51 @@
+        public Game()
+        {
+            // Initliaze the random object
+            rng = new Random();
```
While not incorrect, it would be good practice for all of these variables to have a specified type (so it's easier to read and enforce).
`Random rng = new  Random();`, `Die die1 = new Die(rng);`, etc.

KFANullSense commented on CMP1903_A1_2324/Program.cs:
```diff
@@ -1,19 +1,22 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.CompilerServices;
```
Why do you need this?

Nedkins committed:
Final Code After Review

This is the final version of the code that was used in the video, and has been completed with the help of the reviews.
